src/services/codefixes/fixAddMissingMember.ts

```diff
@@ -465,8 +466,12 @@ namespace ts.codefix {
         const jsxAttributesNode = info.parentDeclaration.attributes;
         const hasSpreadAttribute = some(jsxAttributesNode.properties, isJsxSpreadAttribute);
         const attrs = map(info.attributes, attr => {
-            const value = attr.valueDeclaration ? tryGetValueFromType(context, checker, importAdder, quotePreference, checker.getTypeAtLocation(attr.valueDeclaration)) : createUndefined();
-            return factory.createJsxAttribute(factory.createIdentifier(attr.name), factory.createJsxExpression(/*dotDotDotToken*/ undefined, value));
+            const value = tryGetValueFromType(context, checker, importAdder, quotePreference, checker.getNonMissingTypeOfSymbol(attr));
+            const name = factory.createIdentifier(attr.name);
+            const jsxAttribute = factory.createJsxAttribute(name, factory.createJsxExpression(/*dotDotDotToken*/ undefined, value));
+            // formattingScanner requires the Identifier to have a context for scanning attributes with "-" (data-foo).

```

Comment:
@sandersn Before this fix, there was a crash for names with `-`. In order to scan names `a-b` `formattingScanner` uses `shouldRescanJsxIdentifier` that requires parent. Without `parent`, we will get crash `Error: Debug Failure. False expression: Token end is child end` because `formattingScanner` doesn't fully parse attribute and stops on the first identifier. Does that make sense? Maybe there is a much better solution for that.